src/main/java/fr/openent/mediacentre/service/impl/DefaultFavoriteService.java

```diff
@@ -56,8 +57,27 @@ public void get(String source, String userId, Handler<Either<String, JsonArray>>
     public Future<JsonObject> create(JsonObject favoritesBody) {
         Promise<JsonObject> promise = Promise.promise();
 
-        String errorMessage = "[Mediacentre@DefaultFavoriteService::create] Failed to create favorite in mongo database : ";
-        MongoDb.getInstance().insert(TOKEN_COLLECTION, favoritesBody, MongoDbResult.validResultHandler(FutureHelper.handlerJsonObject(promise, errorMessage)));
+        this.getByMongoId(favoritesBody.getString(_ID, null))
+            .onSuccess(result -> {
+                if (!result.isEmpty()) favoritesBody.remove(_ID);
+                String errorMessage = "[Mediacentre@DefaultFavoriteService::create] Failed to create favorite in mongo database : ";
+                MongoDb.getInstance().insert(TOKEN_COLLECTION, favoritesBody, MongoDbResult.validResultHandler(FutureHelper.handlerJsonObject(promise, errorMessage)));
+            })
+            .onFailure(err -> {
+                String errorMessage = "[Mediacentre@DefaultFavoriteService::create] Failed to check if favorite with same _id already existing in mongo database : ";
```

Comment:
pas moyen de faire un string.format?

```suggestion
 log.error(String.format("[Mediacentre@DefaultFavoriteService::create] Failed to check if favorite with same _id already existing in mongo database : %s", err.getMessage()));
 promise.fail(err.getMessage());
```